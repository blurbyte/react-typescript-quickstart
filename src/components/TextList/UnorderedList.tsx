import styled from '@styles/styledComponents';
import { Color } from '@styles/colors';

export interface UnorderedListProps {
  simple?: boolean;
}

const UnorderedList = styled<UnorderedListProps, 'ul'>('ul')`
  margin: 0;
  padding: ${props => (props.simple ? 0 : 'auto')};
  color: ${Color.DarkGray};
  list-style-type: ${props => (props.simple ? 'none' : 'square')};
`;

export default UnorderedList;
