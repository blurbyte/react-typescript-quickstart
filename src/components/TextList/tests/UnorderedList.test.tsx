import * as React from 'react';
import { shallow } from 'enzyme';

import UnorderedList from '../UnorderedList';

describe('UnorderedList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UnorderedList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly simple list', () => {
    const wrapper = shallow(<UnorderedList simple />);
    expect(wrapper).toMatchSnapshot();
  });
});
