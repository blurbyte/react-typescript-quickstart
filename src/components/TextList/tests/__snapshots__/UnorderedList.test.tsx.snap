// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList renders correctly 1`] = `
.c0 {
  margin: 0;
  padding: auto;
  color: #444444;
  list-style-type: square;
}

<ul
  className="c0"
/>
`;

exports[`UnorderedList renders correctly simple list 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #444444;
  list-style-type: none;
}

<ul
  className="c0"
/>
`;
